edx/edx-app-android
# feat: Add Video Download Qualities option

### Description

[LEARNER-7612](https://openedx.atlassian.net/browse/LEARNER-7612)

### Description

Give learners the choice to download whichever video quality they want to download.

### How to test this PR
- [x] UI: Settings Page
-  Add the video download quality selection option on the Settings screen
- 4 options to select the download quality are available when the user taps into the video download quality selection option
- mobile_low_360p, mobile_high _540p, desktop_mp4 (720p):  all the mappings

- [x]  UI: Videos Page
- A new section will be added to the video downloads page where the user can see the currently selected download quality option
- Tapping on this section will deep-link to the quality selector (1.b.)
- The download size will be calculated based on the selected download quality option
 - [x] Defaults & Persistence 
- The default selection will be ‘Auto’
- Current ‘Auto' behavior is mobile_low → mobile_high → fallback downloads firsts
- Keep this behavior the same for the “auto” selection
- Once the user selects an option, it will persist until the user selects another option. 
- The selected option will be applicable to all courses for that user on that device.
- If the selected quality is not available for a video, the next lower quality will be used to download. If the user has already selected the lowest quality, then we will select the quality that’s closest to that.
- Download videos according to the selected video option

- Tap on video download quality selection on the settings page
- Quality option selected
- Tap on the quality settings on the videos page

| Videos Page Header | Video Quality Dialog | Profile screen VideoQualtiy item |
| ------------- | ------------- | ------------- |
| ![device-2021-09-10-162539](https://user-images.githubusercontent.com/43750646/133776137-03f1c29f-1198-40c4-8954-c839368a4943.png) | ![device-2021-09-14-134106](https://user-images.githubusercontent.com/43750646/133776169-2958284b-34be-49d5-ad15-c851a2c25964.png) | ![device-2021-09-14-134123](https://user-images.githubusercontent.com/43750646/133776753-803f3c41-642e-478d-b5bf-2f39952fba9b.png) |
